gossip/src/duplicate_shred_handler.rs

```diff
@@ -80,6 +82,7 @@ impl DuplicateShredHandler {
             leader_schedule_cache,
             bank_forks,
             duplicate_slots_sender,
+            shred_version,
```

Comment:
I think what you implemented would work, but we could also filter on shred_version inside get_duplicate_shreds(), then DuplicateShredHandler don't need to know about shred_version at all.

@gregcusack what do you think:

Reply:
we don't know shred version until we deser the chunks right?

Reply:
Right now no, to do that we need to add shred_version to chunks, so yes there is a cost.

Err maybe it is better not to touch Gossip message format then.

Reply:
ya, I agree with Wen. looks like we'd have to add shred_version to DuplicateShred which would require some heavier lifting. so current implementation may not be ideal but it is at least much simpler.

Reply:
ya, I agree with Wen. looks like we'd have to add shred_version to DuplicateShred which would require some heavier lifting. so current implementation may not be ideal but it is at least much simpler.